src/pages/instances/forms/DuplicateInstanceSnapshotForm.tsx

```diff
@@ -0,0 +1,318 @@
+import { FC, useState } from "react";
+import { LxdInstance, LxdInstanceSnapshot } from "types/instance";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { createInstance, fetchInstances } from "api/instances";
+import { isClusteredServer } from "util/settings";
+import { useSettings } from "context/useSettings";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+import { fetchClusterMembers } from "api/cluster";
+import { useNavigate } from "react-router-dom";
+import { instanceNameValidation, truncateInstanceName } from "util/instances";
+import { fetchProjects } from "api/projects";
+import { LxdDiskDevice } from "types/device";
+import InstanceLink from "pages/instances/InstanceLink";
+import { useEventQueue } from "context/eventQueue";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  close: () => void;
+}
+
+export interface LxdInstanceSnapshotDuplicate {
+  instanceName: string;
+  targetProject: string;
+  targetClusterMember?: string;
+  targetStoragePool: string;
+  stateful: boolean;
+}
+
+export const duplicateSnapshotPayload = (
```

Comment:
```suggestion
export const instanceFromSnapshotPayload = (
```
We are not duplicating a snapshot.